src/index.ts

```diff
@@ -39,23 +56,33 @@ export async function getPositions({
   })
 }
 
-export async function getTodaysProfits({
+export async function getProfits({
   userAddress,
   filterProtocolIds,
   filterChainIds,
+  toBlockNumbersOverride,
+  filterTimePeriod = TimePeriod.sevenDays,
 }: {
   userAddress: string
+  filterTimePeriod?: TimePeriod
   filterProtocolIds?: Protocol[]
   filterChainIds?: Chain[]
+  toBlockNumbersOverride?: Record<Chain, number | undefined>
 }): Promise<DefiProfitsResponse[]> {
   const runner = async (
     adapter: IProtocolAdapter,
     provider: ethers.JsonRpcProvider,
   ) => {
-    const blockNumber = await provider.getBlockNumber()
-    return await adapter.getOneDayProfit({
+    const chainId = adapter.getProtocolDetails().chainId
+
+    const toBlock =
+      toBlockNumbersOverride?.[chainId] || (await provider.getBlockNumber())
```

Comment:
```suggestion
      toBlockNumbersOverride?.[chainId] ?? (await provider.getBlockNumber())
```
The null coalescing operator is `??`. If your block override was `0`, with `||` it'd be considered invalid and it'll get a new blockNumber.

Ok, I know that chances are the `toBlock` won't be 0, but as a rule of thumb, when you want to do something if the left expression is null or undefined, the correct operator is `??`.